CenterForOpenScience/osf.io
# [PLAT-493] Allow project files to be embedded/rendered in the wiki

Merged and bumped!
<s>❗️ Once the corresponding markdown-it-video PR (https://github.com/brianjgeiger/markdown-it-video/pull/31) is merged, this PR should point to the new version of markdown-it-video before being reviewed/merged ❗️ </s>

## Purpose

This PR allows users to render files from MFR on the wiki editor by safely including an iframe.

## Changes

- This works by forking the "markdown-it-video" plugin and adding a service to it, MFR. The new fork can be found here: https://github.com/Johnetordoff/markdown-it-mfr/tree/add-service-mfr
- To load MFR javascript for iframe sizing changes to the wiki page and project page makos had to be made.

![mfr_embed_better_sizing](https://user-images.githubusercontent.com/9688518/36036043-c6626e54-0d86-11e8-8265-ba7ad3e4e8ee.gif)

## QA Notes

Check that:

- Mfr iframes render when formatted `@[osf](<guid>)` , Note: the guid must be for a file that on prod. - Check for appropriate sizing on different file formats.
- Other video tags like `@[youtube](<id>)` work.

This will also need doc updates at http://help.osf.io.

## Side Effects

- markdown-it-video had a video "doubling" issue discovered while writting this that was wrapped up here: https://github.com/brianjgeiger/markdown-it-video/pull/30

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-493
